src/core/binding.ts

```diff
@@ -86,8 +86,13 @@ export class Binding {
   private willBeInvokedByEvent(event: Event): boolean {
     const eventTarget = event.target
 
-    if (event instanceof KeyboardEvent && this.action.shouldIgnoreKeyboardEvent(event)) {
-      return false
+    if (event.type === "keydown") {
+      // when accepting an autocomplete suggestion, Chrome dispatches a keydown event,
+      // but it isn't a KeyboardEvent instance
+      const keyboardEvent: KeyboardEvent = event instanceof KeyboardEvent ? event : new KeyboardEvent("keydown")
+      if (this.action.shouldIgnoreKeyboardEvent(keyboardEvent)) {
```

Comment:
The underlying [Action.shouldIgnoreKeyboardEvent](https://github.com/hotwired/stimulus/blob/422eb81fa6496d7e24c3983c63e74f3530367cd3/src/core/action.ts#L42-L63) function ultimately [accesses properties on the `KeyboardEvent` instance](https://github.com/hotwired/stimulus/blob/422eb81fa6496d7e24c3983c63e74f3530367cd3/src/core/action.ts#L103), so instantiating a new `KeyboardEvent` instance without those properties might lead to unexpected behavior.

As an alternative, what do you think about using typecasting instead?

```suggestion
     // When accepting an autocomplete suggestion, Chrome dispatches a keydown event that is not a KeyboardEvent instance
    if (event.type === "keydown" && this.action.shouldIgnoreKeyboardEvent(event as KeyboardEvent)) {
```

That assumes that Chrome's synthetic event _does_ have the properties required to behave like a KeyboardEvent. That might not be the case, but my hunch is that it'd behave better as the syntetic event than as an event that Stimulus constructs itself.

Reply:
My intention was to use a newly created keyboard event where all of those keys are blank (so nothing matches). Typecasting probably works, but that synthetic event doesn't have any of the KeyboardEvent properties (besides `type`). The code would be a lot cleaner, though.